apps/src/studio/studio.js

```diff
@@ -5393,9 +5411,14 @@ var checkFinished = function () {
     return true;
   }
 
-  if (Studio.timedOut()) {
-    Studio.result = ResultType.FAILURE;
-    return true;
+  // Don't process timedOut condition here if we have progressConditions to take care of
+  // things, which can include a timedOut.  This avoids having this condition kick in earlier
+  // than level.progressConditions can take care of a timedOut.
+  if (!level.progressConditions) {
+    if (Studio.timedOut()) {
+      Studio.result = ResultType.FAILURE;
+      return true;
```

Comment:
Good question.  In the interest of maintaining legacy compat, I'm going to leave as-is.  tracked.timeOut is set a bit later, in a manner that's compatible with the progressConditions system, but this check still operates on non-progressConditions levels.